### ISSUE TITLE
Folder OAuth 2.0 failed to fetch token

### ISSUE THREAD

**Issue by alphabt** [getinsomnia/insomnia#7596]
### Expected Behavior

Folder's OAuth 2.0 fails to fetch token with "failed to find request"

### Actual Behavior

Folder's OAuth 2.0 able to fetch token.

### Reproduction Steps

1. Create a folder in the scratch pad
2. Go to Auth tab for the folder
3. Switch to OAuth 2.0
4. For the scenario of getting a token as a Microsoft Entra user using the authorization code flow, set the following values:
    - Enabled: [checked]
    - Grant Type: Authorization Code
    - Authorization URL: https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize
    - Access Token URL: https://login.microsoftonline.com/organizations/oauth2/v2.0/token
    - Client ID: [your client id]
    - Redirect URL: http://localhost
    - Scope: [your scope]
    - Credentials: In Request Body
5. Click "Fetch Tokens"
6. Login pops up, sign in using your credential
7. Sign in success, popup closes
8. See "failed to find request"
![image](https://github.com/Kong/insomnia/assets/6440237/8eb57795-2db2-41d0-a7a3-90252f836a50)

Note: This exact auth setup in a request's Auth tab works and I was able to get the access token.

### Is there an existing issue for this?

- [X] I have searched the [issue tracker](https://www.github.com/Kong/insomnia/issues) for this problem.

### Additional Information

Fiddler trace shows requests out to 

GET https://login.microsoftonline.com/organizations/oauth2/v2.0/authorize endpoint returned 200.

### Insomnia Version

 9.3.0-beta.7

### What operating system are you using?

Windows

### Operating System Version

Windows 11 23H2 22631.3810

### Installation method

download from github release

**Comment by jstafford5380** [comment]
Can confirm the same behavior with Okta in the 9.3.0 release. Minor difference in experience, however. Once the sign-in completes in the popup window, it actually logs in to the site. Insomnia doesn't detect any change, so if I go back to Insomnia and press "fetch tokens" again, then I get the "Failed to find request" error. This only seems to happen when using the folder-level auth. It works fine when I do the same setup on an individual request.

```
Version: Insomnia 9.3.0
Build date: 6/27/2024
OS: Windows_NT x64 10.0.19044
Electron: 30.0.9
Node: 20.11.1
Node ABI: 123
V8: 12.4.254.20-electron.0
Architecture: x64
```

**Comment by subnetmarco** [comment]
Prioritizing for 9.3.2 this week.

**Comment by jackkav** [comment]
Looks related to https://github.com/nodejs/node/issues/29481
Will take a look at this.

**Comment by mdsharpe** [comment]
Fixed in 9.3.2 beta 0 for me - thanks!
